File: backend/src/main/scala/services/merchant/objects/StoreStatus.scala
package services.merchant.objects

// Business note: service-owned domain object; mirror it in the frontend only when it is part of the protocol surface.
import system.objects.given
import system.objects.*

import io.circe.{Decoder, Encoder}

enum StoreStatus:
  case Open, Busy, Revoked

object StoreStatus:
  private val enumLabel = text("StoreStatus")
  def render(value: StoreStatus): DisplayText = enumDisplayText(value)
  def parse(value: DisplayText): Option[StoreStatus] = parseEnumValue(value, StoreStatus.values)
  given Encoder[StoreStatus] = enumEncoder
  given Decoder[StoreStatus] = enumDecoder(StoreStatus.values, enumLabel)

File: frontend/src/objects/merchant/StoreStatus.ts
// Business note: frontend mirror of a backend service object; keep names and fields aligned unless this file is explicitly frontend-only.
export const STORE_STATUS = {
  open: 'Open',
  busy: 'Busy',
  revoked: 'Revoked',
} as const

export type StoreStatus = (typeof STORE_STATUS)[keyof typeof STORE_STATUS]
